shaman/controllers/search.py

```diff
@@ -39,6 +38,18 @@ def apply_filters(self, filters):
             query = Repo.filter_by(project=project)
         except KeyError:
             query = Repo.query
+        if filters.get("distros", None):
+            # TODO: we'll need some sort of schema validation here
+            distro_list = util.parse_distro_query(filters.pop("distros"))
+            distro_filter_list = []
+            for distro in distro_list:
+                # for deb-based distros we store codename in the db as version
+                version_filter = distro["distro_codename"]
+                if not distro["distro_codename"]:
+                    # we do not use codenames for rpm-based distros
```

Comment:
once again, the duality of this is biting us back. If we ever get an rpm distro that has codenames and someone wants to use that, we will get into trouble. 

Nothing we can do right now though